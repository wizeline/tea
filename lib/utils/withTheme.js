'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

/* eslint flowtype-errors/enforce-min-coverage: 0 */

var _react = require('react');

var React = _interopRequireWildcard(_react);

var _isEmpty = require('ramda/src/isEmpty');

var _isEmpty2 = _interopRequireDefault(_isEmpty);

var _styledComponents = require('styled-components');

var _defaultTheme = require('../themes/defaultTheme');

var _defaultTheme2 = _interopRequireDefault(_defaultTheme);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } else { var newObj = {}; if (obj != null) { for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) newObj[key] = obj[key]; } } newObj.default = obj; return newObj; } }

var hasThemeProp = function hasThemeProp(props) {
  return props.theme && !(0, _isEmpty2.default)(props.theme);
};

var withPhoenixTheme = function withPhoenixTheme(Component) {
  var result = function result(props) {
    if (hasThemeProp(props)) {
      return React.createElement(Component, props);
    }
    return React.createElement(
      _styledComponents.ThemeProvider,
      { theme: _extends({}, _defaultTheme2.default) },
      React.createElement(Component, props)
    );
  };
  return result;
};

exports.default = withPhoenixTheme;