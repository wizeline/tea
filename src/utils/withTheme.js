// @flow

/* eslint flowtype-errors/enforce-min-coverage: 0 */

import * as React from 'react';
import ThemeProvider from '../ThemeProvider';

const withTheme = (Component: React.ComponentType<Object>) => (
  props: Object,
) => (
  <ThemeProvider>
    <Component {...props} />
  </ThemeProvider>
);

export default withTheme;
