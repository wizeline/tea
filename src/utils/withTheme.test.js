import React from 'react';
import { shallow } from 'enzyme';
import withTheme from './withTheme';

describe('withTheme', () => {
  it('functions correctly without an overriding theme object', () => {
    const Div = () => <div />;
    const ThemedDiv = withTheme(Div);
    const wrapper = shallow(<ThemedDiv />);
    expect(wrapper.html()).toMatchSnapshot();
  });
});
