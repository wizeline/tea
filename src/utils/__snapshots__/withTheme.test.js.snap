// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withTheme functions correctly without an overriding theme object 1`] = `"<div></div>"`;
